app/src/main/java/net/christianbeier/droidvnc_ng/ListenIfAdapter.java

```diff
@@ -0,0 +1,263 @@
+        public TextView txtLabel;
+    }
+
+    // Data to be shown with the adapter
+    private ArrayList<NetworkInterfaceData> data;
+    private int dataSize;
+
+
+    // Some context data for "easy retrieval"
+    private Context mContext;
+    private Resources mResources;
+    private LayoutInflater mInflater;
+
+
+
+
+    public ListenIfAdapter(ArrayList<NetworkInterface> ifs, Context context) {
+        super(context, R.layout.spinner_row, R.id.spinner_text);
+
+        this.mContext = context;
+        this.mResources = this.mContext.getResources();
+        this.mInflater = LayoutInflater.from(this.mContext);
+
+        this.data = new ArrayList<>();
+        this.data.add(NetworkInterfaceData.getAnyOption(this));
+        for (NetworkInterface nic : ifs) {
+            this.data.add(new NetworkInterfaceData(nic, this));
+        }
+        this.dataSize = this.data.size();
+    }
+
+
+
+    public int getItemPositionByIfName(String ifName) {
+        int i = 0;
+        for (NetworkInterfaceData nid : this.data) {
+            if (nid.getName().equals(ifName)) {
+                return i;
+            }
+            i++;
+        }
+
+        return 0;
+    }
+
+
+
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        if (convertView == null) { // Check if view must be recreated, using the famous ViewHolder pattern
+            convertView = this.mInflater.inflate(R.layout.spinner_row, parent, false);
+
+            ViewHolder vh = new ViewHolder();
+            vh.txtLabel = convertView.findViewById(R.id.spinner_text);
+            convertView.setTag(vh);
+        }
+
+        ViewHolder vh = (ViewHolder)convertView.getTag();
+        NetworkInterfaceData nid = this.getItem(position);
+        vh.txtLabel.setText(nid.toString());
+
+        return convertView;
+    }
+
+
+    @Override
+    public NetworkInterfaceData getItem(int position) {
+        if (0 <= position && position < this.getCount()) {
+            return this.data.get(position);
+        }
+        return null;
+    }
+
+
+
+    @Override
+    public int getCount() {
+        return this.dataSize;
+    }
+
+
+
+
+    private String getStringForAny() {
+        return this.mResources.getString(R.string.main_activity_settings_listenif_spin_any);
+    }
+
+
+    /**
+     * Since the majority of the interfaces have standard Linux names, this method tries to get a possible friendly name.
+     * Special thanks go to the pages:
+     * - https://stackoverflow.com/questions/33747787/android-networkinterface-what-are-the-meanings-of-names-of-networkinterface
+     * - https://stackoverflow.com/questions/47488435/meaning-of-network-interface-rmnet-ipa0
+     * for some of the obscure interface names
+     * @param ifName name (technical) of the interface
+     * @return if known, a friendly name for the interface, otherwise, null
+     */
+    private String getFriendlyStringForInterface(String ifName) {
```

Comment:
This manual mapping gets us (or rather me) into a rabbithole of users complaining later on that their exotic device isn't listed. Please just use https://developer.android.com/reference/java/net/NetworkInterface#getDisplayName()

Reply:
I kept a copy of this just in case, I'll consider it solved for now.